state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -2093,6 +2105,50 @@ pub mod pallet {
 				},
 			};
 		}
+
+		pub(super) fn take_refund_fee(
+			total_input_amount: AssetAmount,
+			input_asset: Asset,
+		) -> Result<AssetAmount, DispatchError> {
+			let refund_fee_usdc = MinimumNetworkFeePerChunk::<T>::get();
+
+			let required_refund_fee_as_input_asset = if input_asset == STABLE_ASSET {
+				refund_fee_usdc
+			} else {
+				with_transaction_unchecked(|| {
```

Comment:
Any particular reason for `unchecked`? According to the docs, it only exists for backwards compatibility.

Reply:
Btw. i thought we already had a conversion api for this somewhere? AFAIK it's more accurate since it uses the conversion in the 'right' direction. Here, you'r converting from usdc to the input asset and assuming that the opposite exchange rate is equivalent. This is not necessarily true.

Reply:
Do you mean `calculate_input_for_gas_output`? Seems like currently it is designed for gas specifically, but assuming that gas amounts aren't too different from what we are going to charge here, it should work fine (might need to rename a few things though).